model/src/contract.rs

```diff
@@ -425,25 +470,28 @@ where
     ///
     /// Accessing uninitialized contract state can end in trapping execution
     /// or in the worst case in undefined behaviour.
-    fn deploy_with(&mut self, input: &[u8]) {
+    fn deploy_with(&mut self, input: &[u8]) -> Result<(), RetCode> {
         // Deploys the contract state.
         //
         // Should be performed exactly once during contract lifetime.
         // Consumes the contract since nothing should be done afterwards.
         use ink_core::storage::alloc::Initialize as _;
         self.env.initialize(());
-        let deploy_params = DeployArgs::decode(&mut &input[..]).unwrap();
+        let deploy_params = DeployArgs::decode(&mut &input[..])
+            .map_err(|_err| RetCode::failure())?;
```

Comment:
Do we plan to use the error codes 1-254 so we can debug where the error originates? 

Also can't believe that `unwrap` has been there all along!

Reply:
The `unwrap` should have been an `expect` but it didn't even matter because SRML contracts didn't provide any way to introspect the `expect` message anyway. Now we have better utilities.
We have yet to find out how we are going to handle failures in ink!.